tools/document-pii/src/item_collector.rs

```diff
@@ -0,0 +1,251 @@
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::fs;
+use std::io::BufRead;
+use std::path::{Path, PathBuf};
+
+use crate::EnumOrStruct;
+
+use anyhow::anyhow;
+use syn::punctuated::Punctuated;
+use syn::visit::Visit;
+use syn::{ItemEnum, ItemStruct, UseTree};
+
+/// Iterates over the rust files to collect all the types and use_statements, which is later
+/// used for recursively looking for pii_fields afterwards.
+pub struct ItemCollector {
+    module_path: String,
+    /// Maps from the full path of a type to its AST node.
+    all_types: HashMap<String, EnumOrStruct>,
+    /// Maps from a module_path to all the use_statements and path of local items.
+    use_statements: BTreeMap<String, BTreeSet<String>>,
+}
+
+impl ItemCollector {
+    fn insert_use_statements(&mut self, use_statements: Vec<String>) {
+        self.use_statements
+            .entry(self.module_path.clone())
+            .or_default()
+            .extend(use_statements);
+    }
+
+    /// Gets both a mapping of the full type to a type and its actual AST node, and also the
+    /// use_statements in its module, which is needed to fetch the types that it referes to in its
+    /// fields.
+    #[allow(clippy::type_complexity)]
+    pub fn get_types_and_use_statements(
+        paths: &Vec<PathBuf>,
+    ) -> anyhow::Result<(
+        HashMap<String, EnumOrStruct>,
+        BTreeMap<String, BTreeSet<String>>,
+    )> {
```

Comment:
I agree with clippy here -- I think we could make this method simpler than this return type. If we make a struct with meaningful names, that's already a great start!

Reply:
Ok I did it now, I thought it'd be ugly but then i remembered destructuring structs is possible, so now the original reason for the tuple is gone :P